### DISCUSSION TITLE
eventHandlerClass with initialize Script

### DISCUSSION THREAD

**Discussion by giangianoulas** [eclipse-birt/birt#1519] [Ideas]
Hi

Is there a reason why we cannot both?
As can seen here if we have an initialize script then the handler does not get called.
Where is the harm in this?
```
            org.eclipse.birt.report.engine.script.internal.ReportScriptExecutor#handleInitialize
            if (handleScript((Object)null, scriptExpr, context).didRun()) {
                return;
            }

            IReportEventHandler eh = (IReportEventHandler)getInstance(design, context);
            if (eh != null) {
                eh.initialize(context.getReportContext());
            }
```

**Comment by hvbtup** [comment]
Disclaimer: I never used Java handlers. My personal opinion is:

I don't think that there is a strict *technical* reason for not running both handlers.

OTOH there is no need to allow running both handlers: 
If you alread have developed a Java handler for the item, then what's the reason to put part of the logic into a Javascript handler and run both? It seems safer and cleaner to have all the logic inside the Java handler then.

Probably only a very small percentage of BIRT users know about Java handlers at all, and even less are using them.
So, while changing this behavior should be considered a breaking change in theory, hardly anyone would notice at all.

So while I don't see a reason to change the behavior, I don't have any objections against it.

From my POV: Feel free to submit a PR.
But if you are going to change the behavior, please consider performing this change *for all the other events as well*, not only the initialize event of one special item type (because that would be confusing for everyone except you), and describe the new expected behavior in detail.

**Comment by giangianoulas** [comment]
My sitution is that I inherited a platform with 500+ reports, some use handler others use init script. I am in no situation to know the business logic behind every report. This platform used a 10+ year old birt version so I updated it to 4.12.0, PDF rendering changed a little so i needed to do something about all reports in a single place. I did it with css in the existing java handler.
I already solved the original problem in another way. I will close this but I see no reason for this not to work both ways. I will probably create a PR later. Thank you
